apps/vue-e2e/tests/vue.test.ts

```diff
@@ -3,43 +3,153 @@ import {
-      Test Suites: 1 passed, 1 total
-      Tests:       1 passed, 1 total
-      Snapshots:   0 total
-    `);
+    done();
+  }, 300000);
 
-    const e2eResult = await runNxCommandAsync(`e2e ${appName}-e2e --headless`);
-    expect(e2eResult.stdout).toContain('All specs passed!');
+  describe('--routing', () => {
+    it('should generate app with routing', async (done) => {
+      const appName = uniq('app');
+      ensureNxProject('@nx-plus/vue', 'dist/libs/vue');
+      await runNxCommandAsync(`generate @nx-plus/vue:app ${appName} --routing`);
+
+      await testGeneratedApp(appName, {
+        lint: true,
+        test: false,
+        e2e: true,
+        build: true,
+        buildProd: true,
+      });
+
+      expect(() =>
+        checkFilesExist(
+          `dist/apps/${appName}/js/about.js`,
+          `dist/apps/${appName}/js/about.js.map`
+        )
+      ).not.toThrow();
+
+      done();
+    }, 300000);
+  });
+
+  describe('--style', () => {
+    it('should generate app with scss', async (done) => {
+      const appName = uniq('app');
+      ensureNxProject('@nx-plus/vue', 'dist/libs/vue');
+      await runNxCommandAsync(
+        `generate @nx-plus/vue:app ${appName} --style scss`
+      );
+
+      await testGeneratedApp(appName, {
+        lint: false,
+        test: false,
+        e2e: false,
+        build: true,
+        buildProd: true,
+      });
+
+      done();
+    }, 300000);
+
+    it('should generate app with stylus', async (done) => {
+      const appName = uniq('app');
+      ensureNxProject('@nx-plus/vue', 'dist/libs/vue');
+      await runNxCommandAsync(
+        `generate @nx-plus/vue:app ${appName} --style stylus`
+      );
+
+      await testGeneratedApp(appName, {
+        lint: false,
+        test: false,
+        e2e: false,
+        build: true,
+        buildProd: true,
+      });
+
+      done();
+    }, 300000);
+
+    it('should generate app with less', async (done) => {
+      const appName = uniq('app');
+      ensureNxProject('@nx-plus/vue', 'dist/libs/vue');
+      await runNxCommandAsync(
+        `generate @nx-plus/vue:app ${appName} --style less`
+      );
+
+      await testGeneratedApp(appName, {
+        lint: false,
+        test: false,
+        e2e: false,
+        build: true,
+        buildProd: true,
+      });
+
+      done();
+    }, 300000);
+  });
+
+  describe('vuex', () => {
+    it('should generate app and add vuex', async (done) => {
+      const appName = uniq('app');
+      ensureNxProject('@nx-plus/vue', 'dist/libs/vue');
+      await runNxCommandAsync(`generate @nx-plus/vue:app ${appName}`);
+      await runNxCommandAsync(`generate @nx-plus/vue:vuex ${appName}`);
+
+      await testGeneratedApp(appName, {
+        lint: true,
+        test: false,
+        e2e: false,
+        build: true,
+        buildProd: true,
+      });
+
+      done();
+    }, 300000);
+  });
+
+  it('should generate app with routing and add vuex', async (done) => {
```

Comment:
What's the value of this test? Cover all bases?

Reply:
This test doesn't follow our pattern of testing one option at a time, but I feel the majority of apps will begin with routing and Vuex and there's overlap (e.g. `main.ts`) when they're used together so I wanted to help ensure that combination doesn't break.